### ISSUE TITLE
Benchmarks currently broken

### ISSUE THREAD

**Issue by anntzer** [numpy/numpy#6773]
While trying to improve on #5514, I noticed that as of HEAD the benchmark suite is currently broken.

```
$ python runtests.py --bench bench_core
Building, see build.log...
Build OK
· Discovering benchmarks
·· Error running /usr/bin/python /home/antony/.local/lib/python3.5/site-packages/asv/benchmark.py discover benchmarks /tmp/tmpdz0ljgfl                                      
             STDOUT -------->

             STDERR -------->
             Traceback (most recent call last):
               File "/home/antony/.local/lib/python3.5/site-packages/asv/benchmark.py", line 658, in <module>
                 list_benchmarks(benchmark_dir, fp)
               File "/home/antony/.local/lib/python3.5/site-packages/asv/benchmark.py", line 639, in list_benchmarks
                 for benchmark in disc_benchmarks(root):
               File "/home/antony/.local/lib/python3.5/site-packages/asv/benchmark.py", line 624, in disc_benchmarks
                 for module in disc_files(root, os.path.basename(root) + '.'):
               File "/home/antony/.local/lib/python3.5/site-packages/asv/benchmark.py", line 613, in disc_files
                 module = import_module(package + filename)
               File "/usr/lib/python3.5/importlib/__init__.py", line 126, in import_module
                 return _bootstrap._gcd_import(name[level:], package, level)
               File "<frozen importlib._bootstrap>", line 986, in _gcd_import
               File "<frozen importlib._bootstrap>", line 969, in _find_and_load
               File "<frozen importlib._bootstrap>", line 944, in _find_and_load_unlocked
               File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
               File "<frozen importlib._bootstrap>", line 986, in _gcd_import
               File "<frozen importlib._bootstrap>", line 969, in _find_and_load
               File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
               File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
               File "<frozen importlib._bootstrap_external>", line 662, in exec_module
               File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
               File "/home/antony/src/numpy/benchmarks/benchmarks/__init__.py", line 3, in <module>
                 from . import common
               File "/home/antony/src/numpy/benchmarks/benchmarks/common.py", line 29, in <module>
                 values = [random.uniform(0, 100) for x in range(nx*ny/10)]*10
             TypeError: 'float' object cannot be interpreted as an integer
Traceback (most recent call last):
  File "/home/antony/.local/bin/asv", line 11, in <module>
    sys.exit(main())
  File "/home/antony/.local/lib/python3.5/site-packages/asv/main.py", line 36, in main
    result = args.func(args)
  File "/home/antony/.local/lib/python3.5/site-packages/asv/commands/__init__.py", line 47, in run_from_args
    return cls.run_from_conf_args(conf, args)
  File "/home/antony/.local/lib/python3.5/site-packages/asv/commands/run.py", line 122, in run_from_conf_args
    skip_existing_commits=args.skip_existing_commits
  File "/home/antony/.local/lib/python3.5/site-packages/asv/commands/run.py", line 197, in run
    benchmarks = Benchmarks(conf, regex=bench)
  File "/home/antony/.local/lib/python3.5/site-packages/asv/benchmarks.py", line 285, in __init__
    for benchmark in benchmarks:
  File "/home/antony/.local/lib/python3.5/site-packages/asv/benchmarks.py", line 325, in disc_benchmarks
    dots=False)
  File "/home/antony/.local/lib/python3.5/site-packages/asv/environment.py", line 429, in run
    self._executable] + args, **kwargs)
  File "/home/antony/.local/lib/python3.5/site-packages/asv/util.py", line 431, in check_output
    raise ProcessError(args, retcode, stdout, stderr)
asv.util.ProcessError: Command '/usr/bin/python /home/antony/.local/lib/python3.5/site-packages/asv/benchmark.py discover benchmarks /tmp/tmpdz0ljgfl' returned non-zero exit status 1
```

(whether with asv 0.1.1 or HEAD)

Perhaps more importantly, perhaps travis should be set to run the benchmark suite too (just to be sure it runs)?

**Comment by charris** [comment]
This is a python 3 thing, the bug is using `/` instead of `//` and it wasnt seen because the python file `benchmarks/benchmarks/common.py` is missing the mandatory import line
```
from __future__ import division, absolute_import, print_function
```
We should audit all the new python files and check that they have that line.

**Comment by charris** [comment]
Here is an initial list of files to check
```
benchmarks/benchmarks/common.py
numpy/_build_utils/apple_accelerate.py
numpy/compat/tests/test_compat.py
numpy/core/tests/test_scalarinherit.py
numpy/distutils/msvc9compiler.py
numpy/distutils/msvccompiler.py
numpy/f2py/__main__.py
numpy/lib/tests/test_packbits.py
numpy/linalg/tests/test_deprecations.py
runtests.py
tools/win32build/build-cpucaps.py
```
The import feature needs to be omitted from scripts.

**Comment by pv** [comment]
The `--bench` command setup in `runtests.py` is actually somewhat dodgy, since it assumes the `asv` command launches the same Python version as what's used for `runtests.py`.

**Comment by anntzer** [comment]
sys.executable should solve the issue?

**Comment by pv** [comment]
Unfortunately, there's currently no way to tell `asv` to use an existing environment corresponding to a given python executable.
Workaround would look like this: https://gist.github.com/pv/62a57c7dfb1db8eb04e4

**Comment by pv** [comment]
A simpled alternative would be to just let `asv` do the building itself, as it's done in `--bench-compare`. This requires however that changes are committed before they can be benchmarked.

**Comment by kyleabeauchamp** [comment]
Can someone tell me a python or command line to run the numpy benchmarks with a pre-compiled (conda+MKL) package?  AFAIK, the `runtests` script does not seem to allow combining the `bench` and the `no-build` flags.  And using `np.bench` in an interpreter seems to fail to actually run (report?) the benchmark steps.

TLDR is that I want to do a quick evaluation of some new hardware.

**Comment by charris** [comment]
ISTR that you can use asv directly, but the details escape me. @pv Thoughts?

**Comment by pv** [comment]
benchmarks/README

**Comment by kyleabeauchamp** [comment]
I think this is what I wanted: `asv run --quick --show-stderr --python=same`

**Comment by eric-wieser** [comment]
Pretty sure this was fixed in #6774, right?
